kiwi/repository/apt.py

```diff
@@ -0,0 +1,200 @@
+        and environment
+
+        Attributes
+
+        * :attr:`shared_apt_get_dir`
+            shared directory between image root and build system root
+
+        * :attr:`runtime_apt_get_config_file`
+            apt-get runtime config file name
+
+        * :attr:`apt_get_args`
+            apt-get caller arguments
+
+        * :attr:`command_env`
+            customized os.environ for apt-get
+
+        * :attr:`manager_base`
+            shared location for apt-get repodata between host and image
+
+        :param list custom_args: apt-get arguments
+        """
+        self.custom_args = custom_args
+        if not custom_args:
+            self.custom_args = []
+
+        self.distribution = None
+        self.distribution_path = None
+        self.repo_names = []
+
+        # apt-get support is based on creating a sources file which
+        # contains path names to the repo and its cache. In order to
+        # allow a persistent use of the files in and outside of a chroot
+        # call an active bind mount from RootBind::mount_shared_directory
+        # is expected and required
+        self.manager_base = self.shared_location + '/apt-get'
+
+        self.shared_apt_get_dir = {
+            'sources-dir': self.manager_base + '/sources.list.d',
+            'preferences-dir': self.manager_base + '/preferences.d'
+        }
+
+        self.runtime_apt_get_config_file = NamedTemporaryFile(
+            dir=self.root_dir
+        )
+
+        self.apt_get_args = [
+            '-q', '-c', self.runtime_apt_get_config_file.name, '-y'
+        ] + self.custom_args
+
+        self.command_env = self.__create_apt_get_runtime_environment()
+
+        # config file for apt-get tool
+        self.apt_conf = PackageManagerTemplateAptGet()
+        self.__write_runtime_config()
+
+    def use_default_location(self):
+        """
+        Setup apt-get repository operations to store all data
+        in the default places
+        """
+        self.__write_runtime_config(system_default=True)
+
+    def runtime_config(self):
+        """
+        apt-get runtime configuration and environment
+        """
+        return {
+            'apt_get_args': self.apt_get_args,
+            'command_env': self.command_env,
+            'distribution': self.distribution,
+            'distribution_path': self.distribution_path
+        }
+
+    def add_repo(
+        self, name, uri, repo_type='deb',
+        prio=None, dist=None, components=None
+    ):
+        """
+        Add apt_get repository
+
+        :param string name: repository base file name
+        :param string uri: repository URI
+        :param repo_type: unused
+        :param int prio: unused
+        :param dist: distribution name for non flat deb repos
+        :param components: distribution categories
+        """
+        list_file = self.shared_apt_get_dir['sources-dir'] + \
+            '/' + name + '.list'
```

Comment:
Actually you don't have to stick to 79 per PEP8. :wink: Especially when `\` comes to the deal.